extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -888,10 +888,15 @@ String generateSequenceName(
   @VisibleForTesting
   String generateSequenceName(int groupId)
```

Comment:
How about having this accept TaskGroup object instead of a groupId? Then having a null-check on that argument will achieve much the same thing, but leaves more control in the hands of the caller as to what exact group gets passed in.

It also makes bugs less likely, since a caller must have a real, live TaskGroup object to pass in or else it can't call this method.

Reply:
Do you mean `generateSequenceName` should only accept a non-null `taskGroup` ?
Else it's the same issue, as caller can pass a null. In that case caller will have to check if `groupId` exists in `taskGroups`, and if not decide how to proceed.